File: packages/core/src/model/FlowGraph.ts
import type { FlowDefinition } from './FlowDefinition.js';
import type { FlowEdge } from './FlowEdge.js';
import type { FlowNode } from './FlowNode.js';
import type { ProjectConfig } from './ProjectConfig.js';

export interface FlowGraph {
  id: string;
  name: string;
  flows: FlowDefinition[];
  config?: ProjectConfig;
  /** @deprecated Legacy single-flow shape; migrated via normalizeGraph */
  nodes?: FlowNode[];
  /** @deprecated Legacy single-flow shape; migrated via normalizeGraph */
  edges?: FlowEdge[];
}

File: java/flowcamel-core/src/main/java/com/flowcamel/core/model/FlowGraph.java
package com.flowcamel.core.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.ArrayList;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class FlowGraph {
  public String id = "";
  public String name = "";
  public List<FlowDefinition> flows = new ArrayList<>();
  public ProjectConfig config;
  /** Legacy single-flow; migrated by {@link com.flowcamel.core.graph.GraphNormalizer}. */
  public List<FlowNode> nodes;
  public List<FlowEdge> edges;
}
